src/components/ModularTable/ModularTable.tsx

```diff
@@ -25,21 +25,40 @@ export type Props<D extends Record<string, unknown>> = {
    * Optional extra row to display underneath the main table content.
    */
   footer?: ReactNode;
+  getTableProps?: TablePropGetter<D>;
+  getHeaderProps?: (column: Column<D>) => Record<string, unknown>;
+  getColumnProps?: (column: Column<D>) => Record<string, unknown>;
+  getRowProps?: (column: Column<D>) => Record<string, unknown>;
+  getCellProps?: (cell: Cell<D>) => Record<string, unknown>;
+  getRowId?: UseTableOptions<D>["getRowId"];
 };
 
-function ModularTable({
+const defaultPropGetter = () => ({});
+
+function ModularTable<D extends Record<string, unknown>>({
   data,
   columns,
   emptyMsg,
   footer,
-}: Props<Record<string, unknown>>): JSX.Element {
-  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
-    useTable({ columns, data });
+  getTableProps = defaultPropGetter,
```

Comment:
I think setting default functions for optional function props can be a bit confusing, it makes it harder to understand when you're reading the code further down that the function call is optionally providing additional props.